mmv1/products/vertexai/api.yaml

```diff
@@ -997,3 +999,208 @@ objects:
           The update method to use with this Index. The value must be the followings. If not set, BATCH_UPDATE will be used by default.
           * BATCH_UPDATE: user can call indexes.patch with files on Cloud Storage of datapoints to update.
           * STREAM_UPDATE: user can call indexes.upsertDatapoints/DeleteDatapoints to update the Index and the updates will be applied in corresponding DeployedIndexes in nearly real-time.
+
+# Vertex AI Models
+  - !ruby/object:Api::Resource
+    name: Model
+    base_url: projects/{{project}}/locations/{{location}}/models
+    create_url: projects/{{project}}/locations/{{location}}/models:upload
+    self_link: 'projects/{{project}}/locations/{{location}}/models/{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/vertex-ai/docs'
+      api: 'https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.models'
+    description: "A trained machine learning Model."
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: location
+        description: The location for the resource
+        url_param_only: true
+        required: true
+        input: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
```

Comment:
In Terraform we don't guarantee that resources have a `name` field- let's just make this `modelId`, and remove the associated custom code. You may need to explicitly specify some URLs / links / import formats to do so, as MMv1 *does* make some invalid assumptions about the presence of `name` fields.

Reply:
I don't think this rename makes sense since the `GET` response contains a `name` field.